### PR TITLE
[CARBONDATA-748] put 'if' clause out of the 'for' clause and add test case

### PR DESCRIPTION
**Pull request by mayunSaicmotor**
put the 'if' clause out of the  'for' clause to avoid to execute too many 'if' clause and improve performance

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1092/

File: core/src/test/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImplTest.java [hunk 1]
```diff
@@ -41,21 +41,20 @@ private BitSet setFilterdIndexToBitSetNew(DimensionColumnDataChunk dimensionColu
     BitSet bitSet = new BitSet(numerOfRows);
     if (dimensionColumnDataChunk instanceof FixedLengthDimensionDataChunk) {
       // byte[][] filterValues = dimColumnExecuterInfo.getFilterKeys();
-      for (int i = 0; i < numerOfRows; i++) {
-
-        if (filterValues.length > 1) {
+      if (filterValues.length > 1) {
+        for (int i = 0; i < numerOfRows; i++) {
           int index = CarbonUtil.binarySearch(filterValues, 0, filterValues.length - 1,
```
Comment (chenliang613) [inline core/src/test/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImplTest.java]: Please remove the blank row

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImpl.java [hunk 2]
```diff
@@ -138,23 +138,23 @@ private BitSet setFilterdIndexToBitSet(DimensionColumnDataChunk dimensionColumnD
     BitSet bitSet = new BitSet(numerOfRows);
     if (dimensionColumnDataChunk instanceof FixedLengthDimensionDataChunk) {
       byte[][] filterValues = dimColumnExecuterInfo.getFilterKeys();
-      for (int i = 0; i < numerOfRows; i++) {
+      if (filterValues.length > 1) {
```
Comment (chenliang613) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImpl.java]: Please remove the blank row

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImpl.java [hunk 3]
```diff
@@ -138,23 +138,23 @@ private BitSet setFilterdIndexToBitSet(DimensionColumnDataChunk dimensionColumnD
...
           int index = CarbonUtil.binarySearch(filterValues, 0, filterValues.length - 1,
               dimensionColumnDataChunk.getChunkData(i));
 
           if (index >= 0) {
             bitSet.set(i);
           }
-        } else if (filterValues.length == 1) {
+        }
+      } else if (filterValues.length == 1) {
+
```
Comment (chenliang613) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImpl.java]: Please remove the blank row

File: core/src/test/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImplTest.java (hunk 1 above)
Comment (mayunSaicmotor) [inline core/src/test/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImplTest.java]: removed

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1093/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1094/

File: core/src/test/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImplTest.java [hunk 4]
```diff
@@ -96,13 +93,39 @@ private BitSet setFilterdIndexToBitSet(DimensionColumnDataChunk dimensionColumnD
...
     int filterCnt = 800;
+    comparePerformance(dataCnt, filterCnt, queryCnt, repeatCnt);
+    
+    filterCnt = 100;
+    comparePerformance(dataCnt, filterCnt, queryCnt, repeatCnt);
+
+  }
+  
+  @Test
+  public void testBoundary() {
```
Comment (chenliang613) [inline core/src/test/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImplTest.java]: Can you explain why add testBoundary() ?

Comment (mayunSaicmotor) [inline core/src/test/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImplTest.java]: added test comment

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1096/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1097/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1098/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1099/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**